status/health/window/error_source.go

```diff
@@ -139,23 +135,27 @@ func MustNewAnchoredHealthyIfNotAllErrorsSource(checkType health.CheckType, wind
 // considering health submission rate and window size when using anchored windows.
 // Windows too close to service emission frequency may cause errors to not surface
 func NewAnchoredHealthyIfNotAllErrorsSource(checkType health.CheckType, windowSize time.Duration) (ErrorHealthCheckSource, error) {
-	return newHealthyIfNotAllErrorsSource(checkType, windowSize, true, newOrdinaryTimeProvider())
+	return newHealthyIfNotAllErrorsSource(checkType, windowSize, true, true, newOrdinaryTimeProvider())
 }
 
-func newHealthyIfNotAllErrorsSource(checkType health.CheckType, windowSize time.Duration, useAnchoredWindows bool, timeProvider timeProvider) (ErrorHealthCheckSource, error) {
+func newHealthyIfNotAllErrorsSource(checkType health.CheckType, windowSize time.Duration, useAnchoredWindows bool, requireFirstFullWindow bool, timeProvider timeProvider) (ErrorHealthCheckSource, error) {
```

Comment:
Why make this a configurable boolean if all external constructors set this to true? This should be configurable externally also.

Reply:
They don't all set it as true; NewUnhealthyIfAtLeastOneErrorSource should be come unhealthy for its initial window but uses a wrapped HealthyIfNotAllErrorsSource for its core logic, so calls this with false.